clippy_lints/src/manual_map.rs

```diff
@@ -176,21 +179,38 @@ impl LateLintPass<'_> for ManualMap {
                     } else {
                         ""
                     };
-                    format!(
-                        "|{}{}| {}",
-                        annotation,
-                        some_binding,
-                        snippet_with_context(cx, some_expr.span, expr_ctxt, "..", &mut app).0
-                    )
-                },
+                    if some_expr.needs_unsafe_block {
+                        format!(
+                            "|{}{}| unsafe {{ {} }}",
+                            annotation,
+                            some_binding,
+                            snippet_with_context(cx, some_expr.expr.span, expr_ctxt, "..", &mut app).0
+                        )
+                    } else {
+                        format!(
+                            "|{}{}| {}",
+                            annotation,
+                            some_binding,
+                            snippet_with_context(cx, some_expr.expr.span, expr_ctxt, "..", &mut app).0
+                        )
+                    }
+                }
```

Comment:
These can also be shortened if you add a `let closure_expr = ..` variable for the closure body beforehand.

```rs
let closure_expr = snippet_with_context(cx, some_expr.expr.span, expr_ctxt, "..", &mut app).0;
if some_expr.needs_unsafe_block {
    format!("|{}{}| unsafe {{ {} }}", annotation, some_binding, closure_expr)
} else {
    format!("|{}{}| {}", annotation, some_binding, closure_expr)
}
```

You can add the `closure_expr` variable before the `let body_str = ...` in line 161 as all uses have the same expression.

Reply:
Line 167 (the `then` block of the `if_chain`) doesn't use it.